src/util.c

```diff
@@ -1318,7 +1335,44 @@ static void test_fixedpoint_d2string(void) {
     assert(sz == 0);
 }
 
-#define UNUSED(x) (void)(x)
+#if defined(__linux__)
+/* Since fadvise and mincore is only supported in specific platforms like
+ * Linux, we only verify the fadvise mechanism works in Linux */
+static int cache_exist(int fd) {
+    unsigned char flag;
+    void *m = mmap(NULL, 4096, PROT_READ, MAP_SHARED, fd, 0);
+    assert(m);
+    assert(mincore(m, 4096, &flag) == 0);
+    munmap(m, 4096);
+    /* the least significant bit of the byte will be set if the corresponding
+     * page is currently resident in memory */
+    return flag&1;
+}
+
+static void test_reclaimFilePageCache(void) {
+    char *tmpfile = "/tmp/redis-reclaim-cache-test";
+    int fd = open(tmpfile, O_RDWR|O_CREAT, 0644);
+    assert(fd >= 0);
+
+    /* test write file */
+    char buf[4] = "foo";
+    assert(write(fd, buf, sizeof(buf)) > 0);
+    assert(cache_exist(fd));
+    assert(redis_fsync(fd) == 0);
+    assert(reclaimFilePageCache(fd, 0, 0) == 0);
+    assert(!cache_exist(fd));
```

Comment:
There is also an error here, which can be reproduced on my local valgrind env
https://github.com/redis/redis/actions/runs/4169023701/jobs/7216496035#step:7:1854
```
redis-server: util.c:1378: test_reclaimFilePageCache: Assertion `!cache_exist(fd)' failed.
```

Reply:
i can also reproduce (Linux), it passes when running without valgrind, but fails when running with valgrind.
not sure how we can detect it and skip the test (never done that).

i see there's a RUNNING_ON_VALGRIND macro that requires valgrind.h.
i suppose that this being part of the unit test and not redis itself (something only developers use), it might be a valid solution.

maybe someone knows a better one?

Reply:
I suspect valgrind would do some black magic like mock the mmap syscall to track the memory allocation(as far as I know some allocator claim large trunk of memory form OS with `mmap`), though I'm not an expert on valgrind.

Reply:
let's just skip it with the ugly ifdef

Reply:
to stop the CI failures, i made a PR for this change https://github.com/redis/redis/pull/11808
@uvletter please handle the other issues (test thresholds, and AOF).